neMag/Controllers/PostsController.cs

```diff
@@ -82,9 +81,8 @@ public ActionResult New(Post post, HttpPostedFileBase[] uploadedPhotos)
 
             try
             {
-                // Make sure that a user cannot make multiple reviews on the same product
                 if ((post.isReview &&
-                    !(from p in db.Posts where p.UserId == post.UserId && p.ProductId == post.ProductId && p.isReview select p).Any())
+                    !db.Posts.Where(p => p.UserId == post.UserId && p.ProductId == post.ProductId).Any())
                     || !post.isReview)
```

Comment:
After refactoring, the modified line is missing the _p.isReview_ check, which is important because otherwise a user cannot post a review if he already posted a question.